src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -137,8 +137,9 @@ trait SubmissionComponent {
       def saveSubmissionWorkflowFailures(workflowFailures: Seq[WorkflowFailure]) = {
         DBIO.seq(workflowFailures.map { case (wff) =>
           for {
-            entityId <- loadSubmissionEntityId(workspaceContext.workspaceId, submission.submissionEntity)
-            failureId <- (workflowFailureQuery returning workflowFailureQuery.map(_.id)) += marshalWorkflowFailure(entityId, UUID.fromString(submission.submissionId))
+            wfEntityId <- uniqueResult[EntityRecord](entityQuery.findEntityByName(workspaceContext.workspaceId, wff.entityType, wff.entityName))
```

Comment:
yeah, I can. I was hoping to avoid that refactoring, but I realize what I have is not ideal code. The best place for it would be in workflowFailureQuery.save, to be totally parallel with workflowQuery.save ... but workflowFailureQuery is not yet a first-class object; it's just `protected val workflowFailureQuery = TableQuery[WorkflowFailureTable]`. I'll break it out ... it'll be a bit of a larger change.